main.cpp

```diff
@@ -3697,15 +3697,15 @@ void info_guts(memory_access &raw_access, void *con) {
                 bin = raw_access.read_vector<uint8_t>(raw_access.get_binary_start(), read_size, true);
                 std::unique_ptr<block> first_block = find_first_block(bin, raw_access.get_binary_start());
                 if (first_block) {
-                    // verify stuff
-                    auto all_blocks = get_all_blocks(bin, raw_access.get_binary_start(), first_block, more_cb);
-
                     int block_i = 0;
                     select_group(metadata_info[block_i++], true);
                     info_metadata(first_block.get(), true);
-                    for (auto &block : all_blocks) {
-                        select_group(metadata_info[block_i++], true);
-                        info_metadata(block.get(), true);
+                    if (first_block->next_block_rel != 0) { // if there is more than one block
+                        auto all_blocks = get_all_blocks(bin, raw_access.get_binary_start(), first_block, more_cb);
```

Comment:
Hmmm, does this change imply that if there are two or more blocks, then `get_all_blocks` doesn't return the first block; but if there's only one block, then `get_all_blocks` _does_ return the first block??
Or am I just mis-reading the diff?

Reply:
Yes, `get_all_blocks` returns all other blocks when there are multiple blocks, but it returns the first block again if there is only one block

Reply:
This might be a stupid question (I'm good a those :laughing: ), but I wonder if that suggests that `get_all_blocks` should instead return zero blocks if there is only one block? Seems like it might be more consistent behaviour? :man_shrugging:

Reply:
That's probably for another PR, as I'd need to fix some places it's used so that behaviour works